data/data_generator.jl

```diff
@@ -0,0 +1,582 @@
+using OffsetArrays: Origin
+
+parsehex(str) = parse(UInt32, str, base=16)
+
+function parse_hex_range(line)
+    m = match(r"^([0-9A-F]+)(\.\.([0-9A-F]+))? +; +([^#]+)", line)
+    if isnothing(m)
+        return nothing
+    end
+    i = parsehex(m[1])
+    j = !isnothing(m[3]) ? parsehex(m[3]) : i
+    desc = rstrip(m[4])
+    return (i:j, desc)
+end
+
+function read_hex_ranges(filename)
+    [r for r in parse_hex_range.(readlines(filename)) if !isnothing(r)]
+end
+
+function collect_codepoints(range_desc, description)
+    list = UInt32[]
+    for (r,d) in range_desc
+        if d == description
+            append!(list, r)
+        end
+    end
+    list
+end
+
+#-------------------------------------------------------------------------------
+
+derived_core_properties = read_hex_ranges("DerivedCoreProperties.txt")
+
+ignorable = Set(collect_codepoints(derived_core_properties, "Default_Ignorable_Code_Point"))
+uppercase = Set(collect_codepoints(derived_core_properties, "Uppercase"))
+lowercase = Set(collect_codepoints(derived_core_properties, "Lowercase"))
+
+
+#-------------------------------------------------------------------------------
+function derive_indic_conjunct_break(derived_core_properties)
+    props = Dict{UInt32, String}()
+    for i in collect_codepoints(derived_core_properties, "InCB; Linker")
+        props[i] = "LINKER"
```

Comment:
Can’t this pattern use `setindex!(Ref(props), codepoints, “VALUE”)` here and elsewhere?

Reply:
That works (adding the forgotten `.` to make it a broadcast?) but it seems kind of obscure to me. I'll add a utility function.